### PR TITLE
PHOENIX-2460 Implement scrutiny command to validate whether or not an…

### PR DESCRIPTION
**Pull request by vincentpoon**
… index is in sync with the data table

### CODE REVIEW

File: phoenix-core/src/test/java/org/apache/phoenix/mapreduce/index/TestIndexScrutinyTableOutput.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+public class TestIndexScrutinyTableOutput extends BaseIndexTest {
+
+    private static final long SCRUTINY_TIME_MILLIS = 1502908914193L;
+
+    @Before
+    public void setup() throws Exception {
+        super.setup();
+        conn.createStatement().execute(IndexScrutinyTableOutput.OUTPUT_TABLE_DDL);
+        conn.createStatement().execute(IndexScrutinyTableOutput.OUTPUT_METADATA_DDL);
+        conn.commit();
```
Comment (JamesRTaylor) [inline phoenix-core/src/test/java/org/apache/phoenix/mapreduce/index/TestIndexScrutinyTableOutput.java]: Minor nit: commit() not necessary after DDL

File: phoenix-core/src/test/java/org/apache/phoenix/mapreduce/index/TestIndexScrutinyTableOutput.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import static org.junit.Assert.assertEquals;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+
+import org.apache.phoenix.mapreduce.util.IndexColumnNames;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestIndexScrutinyTableOutput extends BaseIndexTest {
```
Comment (JamesRTaylor) [inline phoenix-core/src/test/java/org/apache/phoenix/mapreduce/index/TestIndexScrutinyTableOutput.java]: Minor nit: Phoenix naming convention (with a few exceptions) is to post fix class with "Test": IndexScrutinyTableOutputTest

File: phoenix-core/src/test/java/org/apache/phoenix/mapreduce/util/TestIndexColumnNames.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import java.sql.SQLException;
+
+import org.apache.phoenix.jdbc.PhoenixConnection;
+import org.apache.phoenix.mapreduce.index.BaseIndexTest;
+import org.apache.phoenix.parse.HintNode.Hint;
+import org.apache.phoenix.schema.PTableKey;
+import org.apache.phoenix.util.QueryUtil;
+import org.junit.Test;
+
+public class TestIndexColumnNames extends BaseIndexTest {
```
Comment (JamesRTaylor) [inline phoenix-core/src/test/java/org/apache/phoenix/mapreduce/util/TestIndexColumnNames.java]: Same here for name: IndexColumnNamesTest

**Comment by vincentpoon** [comment]
@JamesRTaylor thanks for the comments, I now also write the timestamp from the cell to the output table when there's a mismatch.  Also fixed the nits and use md5.  Ran fine on a cluster with 800m rows.

**Comment by stoty** [comment]
Already merged.

**Pull request closed without merging**